android/src/com/mapswithme/maps/editor/PhoneListAdapter.java

```diff
@@ -0,0 +1,156 @@
+
+import com.google.android.material.textfield.TextInputLayout;
+import com.mapswithme.maps.R;
+import com.mapswithme.util.StringUtils;
+import com.mapswithme.util.UiUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class PhoneListAdapter extends RecyclerView.Adapter<PhoneListAdapter.ViewHolder>
+{
+  private List<String> phonesData = new ArrayList<>();
+
+  public PhoneListAdapter()
+  {
+    phonesData.add("");
+  }
+
+  public PhoneListAdapter(String phoneList)
+  {
+    if (TextUtils.isEmpty(phoneList))
+    {
+      phonesData.add("");
+      return;
+    }
+
+    phonesData = new ArrayList<>();
+    for (String p : phoneList.split(";"))
+    {
+      p = p.trim();
+      if (TextUtils.isEmpty(p)) continue;
+      phonesData.add(p);
+    }
+
+    if (phonesData.size() == 0) phonesData.add("");
+  }
+
+  @NonNull
+  @Override
+  public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType)
+  {
+    return new ViewHolder(LayoutInflater.from(parent.getContext())
+                                        .inflate(R.layout.item_phone, parent, false));
+  }
+
+  @Override
+  public void onBindViewHolder(@NonNull PhoneListAdapter.ViewHolder holder, int position)
+  {
+    holder.setPosition(position);
+    holder.setPhone(phonesData.get(position));
+  }
+
+  @Override
+  public int getItemCount()
+  {
+    return phonesData.size();
+  }
+
+  @Override
+  public long getItemId(int position)
+  {
+    return position;
+  }
+
+  public void appendPhone()
+  {
+    phonesData.add("");
+    notifyDataSetChanged();
+  }
+
+  public String getPhoneList()
+  {
+    StringBuilder sb = new StringBuilder();
+    for (int i = 0; i < phonesData.size(); i++)
+    {
+      final String p = phonesData.get(i).trim();
+      if (!TextUtils.isEmpty(p))
+      {
+        if (sb.length() > 0) sb.append(';');
+        sb.append(p);
+      }
+    }
+    return sb.toString();
+  }
+
+  protected class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener
+  {
+    private int mPosition = -1;
+    private final EditText mInput;
+    private final ImageView deleteButton;
+
+    public ViewHolder(@NonNull View itemView)
+    {
+      super(itemView);
+
+      mInput = itemView.findViewById(R.id.input);
+      final TextInputLayout phoneInput = itemView.findViewById(R.id.phone_input);
+      mInput.addTextChangedListener(new StringUtils.SimpleTextWatcher()
+      {
+        @Override
+        public void onTextChanged(CharSequence s, int start, int before, int count)
+        {
+          UiUtils.setInputError(phoneInput, Editor.nativeIsPhoneValid(s.toString()) ? 0 : R.string.error_enter_correct_phone);
+          PhoneListAdapter.this.updatePhoneItem(mPosition, mInput.getText().toString());
+        }
+      });
+
+      deleteButton = itemView.findViewById(R.id.delete_icon);
+      deleteButton.setOnClickListener(this);
+      //TODO: setting icons from code because icons defined in layout XML are white.
```

Comment:
How to properly define icons from XML, so they will work in both light and dark themes?

Reply:
nit: we insert spaces after slashes.